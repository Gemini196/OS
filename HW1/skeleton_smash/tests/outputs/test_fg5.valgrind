==41153== Memcheck, a memory error detector
==41153== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==41153== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==41153== Command: /home/student/Desktop/OS/HW1/skeleton_smash/smash
==41153== Parent PID: 41149
==41153== 
==41153== Invalid read of size 1
==41153==    at 0x5488B30: __strcmp_sse2_unaligned (strcmp-sse2-unaligned.S:24)
==41153==    by 0x10D95B: QuitCommand::execute() (Commands.cpp:639)
==41153==    by 0x10F240: SmallShell::executeCommand(char const*) (Commands.cpp:958)
==41153==    by 0x110AA6: main (smash.cpp:51)
==41153==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==41153== 
==41153== 
==41153== Process terminating with default action of signal 11 (SIGSEGV)
==41153==  Access not within mapped region at address 0x0
==41153==    at 0x5488B30: __strcmp_sse2_unaligned (strcmp-sse2-unaligned.S:24)
==41153==    by 0x10D95B: QuitCommand::execute() (Commands.cpp:639)
==41153==    by 0x10F240: SmallShell::executeCommand(char const*) (Commands.cpp:958)
==41153==    by 0x110AA6: main (smash.cpp:51)
==41153==  If you believe this happened as a result of a stack
==41153==  overflow in your program's main thread (unlikely but
==41153==  possible), you can try to increase the size of the
==41153==  main thread stack using the --main-stacksize= flag.
==41153==  The main thread stack size used in this run was 8388608.
==41153== 
==41153== FILE DESCRIPTORS: 6 open at exit.
==41153== Open file descriptor 1: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_fg5.out
==41153==    at 0x54EF957: dup2 (syscall-template.S:78)
==41153==    by 0x1109FD: main (smash.cpp:41)
==41153== 
==41153== Open file descriptor 0:
==41153==    at 0x54EF957: dup2 (syscall-template.S:78)
==41153==    by 0x1109AB: main (smash.cpp:33)
==41153== 
==41153== Open file descriptor 5:
==41153==    at 0x54EF927: dup (syscall-template.S:78)
==41153==    by 0x110B4B: __static_initialization_and_destruction_0(int, int) (smash.cpp:9)
==41153==    by 0x110B67: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==41153==    by 0x110BEC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==41153==    by 0x5400C17: (below main) (libc-start.c:266)
==41153== 
==41153== Open file descriptor 4: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_fg5.out
==41153==    at 0x54EF927: dup (syscall-template.S:78)
==41153==    by 0x110B3B: __static_initialization_and_destruction_0(int, int) (smash.cpp:8)
==41153==    by 0x110B67: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==41153==    by 0x110BEC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==41153==    by 0x5400C17: (below main) (libc-start.c:266)
==41153== 
==41153== Open file descriptor 3: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_fg5.valgrind
==41153==    <inherited from parent>
==41153== 
==41153== Open file descriptor 2: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_fg5.err
==41153==    <inherited from parent>
==41153== 
==41153== 
==41153== HEAP SUMMARY:
==41153==     in use at exit: 869 bytes in 13 blocks
==41153==   total heap usage: 39 allocs, 26 frees, 83,289 bytes allocated
==41153== 
==41153== 5 bytes in 1 blocks are still reachable in loss record 1 of 9
==41153==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==41153==    by 0x10B299: _parseCommandLine(char const*, char**) (Commands.cpp:47)
==41153==    by 0x10D93E: QuitCommand::execute() (Commands.cpp:637)
==41153==    by 0x10F240: SmallShell::executeCommand(char const*) (Commands.cpp:958)
==41153==    by 0x110AA6: main (smash.cpp:51)
==41153== 
==41153== 16 bytes in 1 blocks are still reachable in loss record 2 of 9
==41153==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==41153==    by 0x10EC35: SmallShell::SmallShell() (Commands.cpp:901)
==41153==    by 0x1108A0: SmallShell::getInstance() (Commands.h:235)
==41153==    by 0x110970: main (smash.cpp:21)
==41153== 
==41153== 24 bytes in 1 blocks are still reachable in loss record 3 of 9
==41153==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==41153==    by 0x10E017: JobsList::JobsList() (Commands.cpp:725)
==41153==    by 0x10EC40: SmallShell::SmallShell() (Commands.cpp:901)
==41153==    by 0x1108A0: SmallShell::getInstance() (Commands.h:235)
==41153==    by 0x110970: main (smash.cpp:21)
==41153== 
==41153== 24 bytes in 1 blocks are still reachable in loss record 4 of 9
==41153==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==41153==    by 0x10F07E: SmallShell::CreateCommand(char const*) (Commands.cpp:939)
==41153==    by 0x10F19A: SmallShell::executeCommand(char const*) (Commands.cpp:946)
==41153==    by 0x110AA6: main (smash.cpp:51)
==41153== 
==41153== 72 bytes in 3 blocks are still reachable in loss record 5 of 9
==41153==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==41153==    by 0x110584: __gnu_cxx::new_allocator<std::_List_node<JobsList::JobEntry*> >::allocate(unsigned long, void const*) (new_allocator.h:111)
==41153==    by 0x1104CA: std::allocator_traits<std::allocator<std::_List_node<JobsList::JobEntry*> > >::allocate(std::allocator<std::_List_node<JobsList::JobEntry*> >&, unsigned long) (alloc_traits.h:436)
==41153==    by 0x11036A: std::__cxx11::_List_base<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_get_node() (stl_list.h:383)
==41153==    by 0x110189: std::_List_node<JobsList::JobEntry*>* std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_create_node<JobsList::JobEntry* const&>(JobsList::JobEntry* const&) (stl_list.h:572)
==41153==    by 0x10FF6D: void std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_insert<JobsList::JobEntry* const&>(std::_List_iterator<JobsList::JobEntry*>, JobsList::JobEntry* const&) (stl_list.h:1801)
==41153==    by 0x10FBA7: std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::push_back(JobsList::JobEntry* const&) (stl_list.h:1118)
==41153==    by 0x10E12B: JobsList::addJob(char const*, int, long, bool) (Commands.cpp:740)
==41153==    by 0x10BA02: ExternalCommand::execute() (Commands.cpp:162)
==41153==    by 0x10F240: SmallShell::executeCommand(char const*) (Commands.cpp:958)
==41153==    by 0x110AA6: main (smash.cpp:51)
==41153== 
==41153== 160 bytes in 1 blocks are still reachable in loss record 6 of 9
==41153==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==41153==    by 0x10D8F6: QuitCommand::execute() (Commands.cpp:632)
==41153==    by 0x10F240: SmallShell::executeCommand(char const*) (Commands.cpp:958)
==41153==    by 0x110AA6: main (smash.cpp:51)
==41153== 
==41153== 168 bytes in 3 blocks are still reachable in loss record 7 of 9
==41153==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==41153==    by 0x10E0D7: JobsList::addJob(char const*, int, long, bool) (Commands.cpp:738)
==41153==    by 0x10BA02: ExternalCommand::execute() (Commands.cpp:162)
==41153==    by 0x10F240: SmallShell::executeCommand(char const*) (Commands.cpp:958)
==41153==    by 0x110AA6: main (smash.cpp:51)
==41153== 
==41153== 200 bytes in 1 blocks are still reachable in loss record 8 of 9
==41153==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==41153==    by 0x10EC52: SmallShell::SmallShell() (Commands.cpp:902)
==41153==    by 0x1108A0: SmallShell::getInstance() (Commands.h:235)
==41153==    by 0x110970: main (smash.cpp:21)
==41153== 
==41153== 200 bytes in 1 blocks are still reachable in loss record 9 of 9
==41153==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==41153==    by 0x10D904: QuitCommand::execute() (Commands.cpp:633)
==41153==    by 0x10F240: SmallShell::executeCommand(char const*) (Commands.cpp:958)
==41153==    by 0x110AA6: main (smash.cpp:51)
==41153== 
==41153== LEAK SUMMARY:
==41153==    definitely lost: 0 bytes in 0 blocks
==41153==    indirectly lost: 0 bytes in 0 blocks
==41153==      possibly lost: 0 bytes in 0 blocks
==41153==    still reachable: 869 bytes in 13 blocks
==41153==         suppressed: 0 bytes in 0 blocks
==41153== 
==41153== For counts of detected and suppressed errors, rerun with: -v
==41153== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
