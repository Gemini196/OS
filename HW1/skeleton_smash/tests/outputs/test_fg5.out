smash> submitted by Eran
smash> smash> smash> smash> sleep 200& : 46989
smash: got ctrl-Z
smash: process 46989 was stopped
smash> [1] sleep 100& : 46988 2 secs
[3] sleep 300& : 46990 2 secs
[4] sleep 200& : 46989 0 secs (stopped)
smash> 